### PR TITLE
Add yang model for metadata semantics

### PR DESCRIPTION
**Pull request by mickeyspiegel**

### CODE REVIEW

**Comment by mickeyspiegel** [comment]
We need to decide whether there should be a separate model just for metadata semantics between nodes and monitors, or whether the yang model should cover all of data plane telemetry including normal management and configuration.

In openconfig, they strongly advocate a separation of config and operational state. If the scope remains narrow, then we won't have any config state, only operational state, so it deviates from openconfig style to some extent.

File: telemetry/code/models/p4-dtel-metadata-semantics.yang [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+    leaf size-units {
+      type identityref {
+        base METADATA_SIZE_UNITS;
+      }
+      description
+        "Units of size";
+    }
+
+    leaf cell-size {
+      when "../size-units = 'METADATA_CELLS'" {
```
Comment (chrispsommers) [inline telemetry/code/models/p4-dtel-metadata-semantics.yang:186]: Shouldn't this 'when' be around the entire leaf? I'm not sure why you'd make the description conditional, especially when there is another description below.

Comment (mickeyspiegel) [reply telemetry/code/models/p4-dtel-metadata-semantics.yang:186]: From RFC 6020:
   The "when" statement makes its parent data definition statement
   conditional.  The node defined by the parent data definition
   statement is only valid when the condition specified by the "when"
   statement is satisfied.

Comment (chrispsommers) [reply telemetry/code/models/p4-dtel-metadata-semantics.yang:186]: Got it, thanks. My yang is rusty!

**Review by mhira1 — APPROVED** [review]
Mickey, this is a good initial version of the model. Please go ahead and commit it, other contributors can refine further. As we discussed at the working group meeting, please capture the open items from discussions in working group meetings - either as comments in the model or as a separate file in the repository, as you feel appropriate.

**Pull request merged**